pkg/api/norman/customization/cred/store.go

```diff
@@ -3,22 +3,30 @@ package cred
 import (
 	"encoding/base64"
 	"fmt"
+	"strconv"
 	"strings"
+	"time"
 
 	"github.com/rancher/norman/httperror"
 	"github.com/rancher/norman/store/transform"
 	"github.com/rancher/norman/types"
 	"github.com/rancher/norman/types/convert"
+	"github.com/rancher/norman/types/values"
 	"github.com/rancher/rancher/pkg/api/norman/customization/namespacedresource"
+	apimgmtv3 "github.com/rancher/rancher/pkg/apis/management.cattle.io/v3"
 	"github.com/rancher/rancher/pkg/controllers/provisioningv2/cluster"
 	provv1 "github.com/rancher/rancher/pkg/generated/controllers/provisioning.cattle.io/v1"
 	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
 	v3 "github.com/rancher/rancher/pkg/generated/norman/management.cattle.io/v3"
 	"github.com/rancher/rancher/pkg/namespace"
+	"gopkg.in/yaml.v3"
 	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/utils/ptr"
 )
 
-func Wrap(store types.Store, ns v1.NamespaceInterface, nodeTemplateLister v3.NodeTemplateLister, provClusterCache provv1.ClusterCache) types.Store {
+const CloudCredentialExpirationAnnotation = "cattle.io/expiration-date"
```

Comment:
Overall change LGTM. I just think the annotation name `"cattle.io/expiration-date"` is a bit misleading since we're actually storing a Unix timestamp instead. But if the UI team is already on board with this name and using it, I wouldn't hold up the merge over this.

Reply:
It's still open to being changed AFAIK, do you have any suggestions for a better name?

Reply:
Don't have a perfect replacement but I’m thinking `expiration-timestamp`/`expiration-time` or (as Sergey mentioned in the issue description) `expiry-time` might be more apt. We could also consider `expires-at` similar to how we have `ExpiresAt` field for tokens.